### PR TITLE
Enable ALLOW_TARGET_PLATFORM_SPECIALIZATION

### PR DESCRIPTION
**Pull request by ksuther**
### New Pull Request Checklist

- [x] I have read and understood the [CONTRIBUTING guide](https://github.com/CocoaLumberjack/CocoaLumberjack/blob/master/.github/CONTRIBUTING.md)
- [x] I have read the [Documentation](http://cocoadocs.org/docsets/CocoaLumberjack/)
- [x] I have searched for a similar pull request in the [project](https://github.com/CocoaLumberjack/CocoaLumberjack/pulls) and found none

<br/>

- [x] I have updated this branch with the latest master to avoid conflicts (via merge from master or rebase)
- [ ] I have added the required tests to prove the fix/feature I am adding
- [ ] I have updated the documentation (if necessary)
- [x] I have run the tests and they pass
- [x] I have run the lint and it passes (`pod lib lint`)

This merge request fixes / refers to the following issues: ...

### Pull Request Description

This allows the framework to build for multiple platforms which fixes errors building when a project uses CocoaLumberjack in both iOS and watchOS targets.

Without this set we get errors when building with CocoaLumberjack added as a dependency directly through Xcode (rather than using pods, Carthage, or SPM).

### CODE REVIEW

**Review by sushichop — APPROVED** [review]
LGTM👍 @ksuther

File: Configs/Module-Shared.xcconfig [hunk 1]
```diff
@@ -230,6 +230,7 @@ SKIP_INSTALL = YES
 
 // The list of supported platforms from which a base SDK can be used. This setting is used if the product can be built for multiple platforms using different SDKs.
 SUPPORTED_PLATFORMS = macosx iphoneos iphonesimulator watchsimulator watchos appletvsimulator appletvos
+ALLOW_TARGET_PLATFORM_SPECIALIZATION = YES
```
Comment (ffried) [inline Configs/Module-Shared.xcconfig:233]: Small Nitpick:
We sort the settings alphabetically in this file for easier lookup. Can you move this one to the beginning please?
Also, add a small comment above it, what exactly it does.

**Review by ffried — CHANGES_REQUESTED** [review]
Looks good. Just a small nitpick from my side.

File: Configs/Module-Shared.xcconfig (hunk 1 above)
Comment (ksuther) [reply Configs/Module-Shared.xcconfig]: Done, thanks!

**Pull request merged**